601-spring-data-primitive-types/src/test/java/io/quarkus/qe/spring/data/CommonsHeadersTest.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.qe.spring.data;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.http.Headers;
+import io.restassured.response.Response;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+
+import static io.restassured.RestAssured.when;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+
+@QuarkusTest
+public class CommonsHeadersTest {
+    // QUARKUS-547
+    @Test
+    @DisplayName("prototype scope: x-count header must be equal to request amount")
+    void testRequestAmount(){
+        Set<String> xCount = new HashSet<>();
+
+        for(int index = 0; index < 3; index++){
+            Headers headers =  when().get("/cat/customFindDistinctivePrimitive/2").headers();
+            xCount.addAll(new HashSet<>(headers.getValues("x-count")));
+        }
+
+        assertThat("Unexpected x-count header value(Spring Prototype Scope). Must be the same as HTTP request amount.", xCount.size(), is(3));
+    }
+    // QUARKUS-547
+    @Test
+    @DisplayName("Singleton scope: x-instance-id header must be the same")
+    void testInstanceId(){
+        Set<String> instanceIds = new HashSet<>();
+        for(int index = 0; index < 3; index++){
+            Headers headers =  when().get("/cat/customFindDistinctivePrimitive/2").headers();
+            instanceIds.addAll(new HashSet<>(headers.getValues("x-instance")));
+        }
+
+        assertThat("Unexpected x-instance header value(Spring Singleton Scope). Must be 1", instanceIds.size(), is(1));
+    }
+    // QUARKUS-547
+    @Test
+    @DisplayName("request scope: x-request header must be equal to request amount")
+    public void testRequestScope() {
+        Set<String> requestIds = new HashSet<>();
+        for(int index = 0; index < 3; index++){
+            Headers headers =  when().get("/cat/customFindDistinctivePrimitive/2").headers();
+            requestIds.addAll(new HashSet<>(headers.getValues("x-request")));
```

Comment:
Why not adding the collection to the set directly? requestIds.addAll(headers.getValues...)